sqlglot/dialects/dialect.py

```diff
@@ -275,7 +275,7 @@ def no_tablesample_sql(self: Generator, expression: exp.TableSample) -> str:
 
 def no_pivot_sql(self: Generator, expression: exp.Pivot) -> str:
     self.unsupported("PIVOT unsupported")
-    return self.sql(expression)
+    return ""
```

Comment:
This fixes an infinite loop when `self.unsupported` doesn't raise.

Btw, it seems like DuckDB [does support a PIVOT statement](https://duckdb.org/docs/sql/statements/pivot.html), although its syntax seems a bit different (although they say they also support the "standard" PIVOT syntax?). I tried some examples locally but couldn't get them to work, so not sure what's up with that. In any case, we can tackle this in a separate PR.

Reply:
shouldn't it be super() ?

Reply:
i'm sure many other dialects don't support pivots so this is fine

Reply:
This can't be done without running the optimizer and with the current approach we don't transform them to CASEs